node/clients/checkout.ts

```diff
@@ -0,0 +1,215 @@
+import { AuthenticationError, ForbiddenError, InstanceOptions, IOContext, JanusClient, RequestConfig, UserInputError } from '@vtex/api'
+import { AxiosError } from 'axios'
+
+const statusToError = (e: any) => {
+  if (!e.response) {
+    throw e
+  }
+  const { response } = e as AxiosError
+  const { status } = response!
+  if (status === 401) {
+    throw new AuthenticationError(e)
+  }
+  if (status === 403) {
+    throw new ForbiddenError(e)
+  }
+  if (status === 400) {
+    throw new UserInputError(e)
+  }
+  throw e
+}
+
+export interface SimulationData {
+  country: string
+  items: any[]
+  postalCode: string
+}
+
+export class Checkout extends JanusClient {
+  constructor (ctx: IOContext, options?: InstanceOptions) {
+    super(ctx, {
+      ...options,
+      headers: {
+        ...options && options.headers,
+        ...ctx.adminUserAuthToken ? {VtexIdclientAutCookie: ctx.adminUserAuthToken} : null,
+        ...ctx.storeUserAuthToken ? {[`VtexIdclientAutCookie_${ctx.account}`]: ctx.storeUserAuthToken} : null,
+      }
+    })
+  }
+
+  private getCommonHeaders = () => {
+    const { orderFormId } = this.context as CustomIOContext
+    const checkoutCookie = orderFormId ? `checkout.vtex.com=__ofid=${orderFormId};` : ''
+    return {
+      Cookie: `${checkoutCookie};vtex_segment=${this.context.segmentToken};vtex_session=${this.context.sessionToken};`,
+    }
+  }
+
+  private getChannelQueryString = () => {
+    const { segment } = this.context as CustomIOContext
+    const channel = segment && segment.channel
+    const queryString = channel ? `?sc=${channel}` : ''
```

Comment:
Same exact thing needs to be ported to catalog client, this is giving @lowbeer a headache! Can you help him?